docs/source/execute_api.rst

```diff
@@ -0,0 +1,143 @@
+Executing notebooks
+===================
```

Comment:
I think that this file conveys that manually creating an execute preprocessor is **the way** to execute notebook, which is not false, but not technically the canonical way. We have a special Exporter :`from nbconvert.exporters import NotebookExporter` which is made for notebook to notebook conversion (it does extra things). 

The canonical way would be

```python
from nbconvert.exporters import NotebookExporter, HTMLExporter
from traitlets.config import Config

c = Config({'ExecutePreprocessor':{'enable':True}})
E = NotebookExporter(config=c) # or E = HTMLExporter(config=c)

notebook, resources = E.from_filename('0 - notebook.ipynb')
# TODO save notebook
```

One of the big advantage of this approach is that it's identical wether you convert to HTML, rst, Latex, you "just" have to change the exporter you use. 

The other advantage is that it allows to imedialelty now how to do the same things for others and multiple preprocessors, in particular the following that all exporters inherit from : 

```python
List([
                                  'nbconvert.preprocessors.ClearOutputPreprocessor',
                                  'nbconvert.preprocessors.ExecutePreprocessor',
                                  'nbconvert.preprocessors.coalesce_streams',
                                  'nbconvert.preprocessors.SVG2PDFPreprocessor',
                                  'nbconvert.preprocessors.CSSHTMLHeaderPreprocessor',
                                  'nbconvert.preprocessors.LatexPreprocessor',
                                  'nbconvert.preprocessors.HighlightMagicsPreprocessor',
                                  'nbconvert.preprocessors.ExtractOutputPreprocessor',
                              ],
```

You "just" have to enable the preprocessors you use

The exporter should also take care of having the export process stateless, which make the conversion less prone to error. 

While the "how to use preprocessor" is nice, I think it should mention that unless you know what you are doing, using preprocessor directly is likely not what you want to do.